alvr/client_core/src/platform/android/decoder.rs

```diff
@@ -242,55 +259,86 @@ pub fn video_decoder_split(
+                {
+                    let mut decoder_lock = decoder_sink.lock();
 
-                decoder_ready_notifier.notify_one();
-            }
+                    *decoder_lock = Some(Arc::clone(&decoder));
 
-            while running.value() {
-                match decoder.dequeue_output_buffer(Duration::from_millis(1)) {
-                    Ok(DequeuedOutputBufferInfoResult::Buffer(buffer)) => {
-                        // The buffer timestamp is actually nanoseconds
-                        let presentation_time_ns = buffer.info().presentation_time_us();
+                    decoder_ready_notifier.notify_one();
+                }
 
-                        if let Err(e) =
-                            decoder.release_output_buffer_at_time(buffer, presentation_time_ns)
-                        {
-                            error!("Decoder dequeue error: {e}");
+                while running.value() {
+                    match decoder.dequeue_output_buffer(Duration::from_millis(1)) {
+                        Ok(DequeuedOutputBufferInfoResult::Buffer(buffer)) => {
+                            // The buffer timestamp is actually nanoseconds
+                            let presentation_time_ns = buffer.info().presentation_time_us();
+
+                            if let Err(e) =
+                                decoder.release_output_buffer_at_time(buffer, presentation_time_ns)
+                            {
+                                error!("Decoder dequeue error: {e}");
+                            }
                         }
-                    }
-                    Ok(DequeuedOutputBufferInfoResult::TryAgainLater) => thread::yield_now(),
-                    Ok(i) => info!("Decoder dequeue event: {i:?}"),
-                    Err(e) => {
-                        error!("Decoder dequeue error: {e}");
+                        Ok(DequeuedOutputBufferInfoResult::TryAgainLater) => thread::yield_now(),
+                        Ok(i) => info!("Decoder dequeue event: {i:?}"),
+                        Err(e) => {
+                            error!("Decoder dequeue error: {e}");
 
-                        // lessen logcat flood (just in case)
-                        thread::sleep(Duration::from_millis(50));
+                            // lessen logcat flood (just in case)
+                            thread::sleep(Duration::from_millis(50));
+                        }
                     }
                 }
-            }
 
-            // Destroy all resources
-            decoder_sink.lock().take(); // Make sure the shared ref is deleted first
-            decoder.stop().unwrap();
-            drop(decoder);
+                // Destroy all resources
+                decoder_sink.lock().take(); // Make sure the shared ref is deleted first
+                decoder.stop().unwrap();
+                drop(decoder);
+            } else {
+                show_err(preparing_decoder);
```

Comment:
Don't try to print the whole object. Rather, if you handle both Ok and Err cases, prefer using match instead of if-let, then print only the error in the Err branch.